semanticdb-gradle-plugin/src/main/scala/SemanticdbGradlePlugin.scala

```diff
@@ -396,8 +400,50 @@ class WriteDependencies extends DefaultTask {
       .foreach(path => java.nio.file.Files.createDirectories(path.getParent()))
 
     val deps = List.newBuilder[String]
+    val project = getProject()
+
+    // List the project itself as a dependency so that we can assign project name/version to symbols that are defined in this project.
+    // The code below is roughly equivalent to the following with Groovy:
+    //   deps += "$publication.groupId $publication.artifactId $publication.version $sourceSets.main.output.classesDirectory"
+    try {
+      for {
+        classesDirectory <- project
+          .getExtensions()
+          .getByType(classOf[SourceSetContainer])
+          .getByName("main")
```

Comment:
I guess to handle test sources here we'd need to modify the artifact id or something, given that the descriptor doesn't have a notion of classifier like maven dependencies have

Reply:
Is it common to publish test sources? We can support the test scope as long as there is a reliable way to know the artifact ID of the test config

Reply:
I've seen set ups where a library publishes a testkit from its tests, but now that I think about it - it's frequently discouraged and doesn't work with the rest of tooling: https://github.com/scalacenter/bloop/pulls?q=is%3Apr+author%3Akeynmol+is%3Aclosed+test-jar

So let's ignore this for now until there's a clear need.

Reply:
We can probably support it if there's demand and there's support to extract the right information